### PR TITLE
Fixes reagent transfer logging on organs

### LINKED ISSUE #80907: Force-feeding somebody a pill does not generate useful attack logs
Reporting client version: 514.1589

## Issue Summary
Clicking somebody with a pill to feed it to them generates one attack log on yourself indicating that you transferred reagents to a [stomach]. It is identical to feeding yourself a pill and does not indicate a target. No attack log is generated on the target.

These logs work properly for feeding somebody a food item, forcing them to drink from a beaker or other reagent container, and applying medical patches. The pill is the only broken one I can find. This issue report was prompted by a griefing chemist.

## Round ID:
[222000](https://scrubby.melonmesa.com/round/222000)

## Testmerges:
- [Ref Tracking (Live)](https://www.github.com/tgstation/tgstation/pull/80879)
- [Optimizes Reftracking (Bigly) (Plus harddel fixes)](https://www.github.com/tgstation/tgstation/pull/80443)

## Reproduction:
1. eat a pill
2. force-feed somebody a pill
3. see two identical attack logs on yourself and nothing on the victim
4. confuse admins
image
image
The bottom image shows useful admin logs for force-feeding/applying other items, unlike the pill.

### PR DESCRIPTION
**Pull request by lessthnthree**
## About The Pull Request

Fixes combat log for reagent transfer into organs, closes https://github.com/tgstation/tgstation/issues/80907

## Changelog
Not player-facing.

### CODE REVIEW

File: code/modules/reagents/chemistry/holder/holder.dm [hunk 1]
```diff
@@ -532,9 +532,14 @@
 		remove_reagent(reagent.type, transfer_amount)
 		transfer_log[reagent.type] = list(REAGENT_TRANSFER_AMOUNT = transfer_amount, REAGENT_PURITY = reagent.purity)
 
-	if(transferred_by && target_atom)
-		target_atom.add_hiddenprint(transferred_by) //log prints so admins can figure out who touched it last.
-		log_combat(transferred_by, target_atom, "transferred reagents ([get_external_reagent_log_string(transfer_log)]) from [my_atom] to")
+	if(transferred_by)
```
Comment (LemonInTheDark) [inline code/modules/reagents/chemistry/holder/holder.dm:535]: you dropped a check here, was that intentional? when would we want to log combat with no target atom?

File: code/modules/reagents/chemistry/holder/holder.dm [hunk 2]
```diff
@@ -532,9 +532,14 @@
...
+		if(isorgan(target_atom))
+			var/obj/item/organ/target_organ = target_atom
+			target_organ.owner?.add_hiddenprint(transferred_by) //log prints so admins can figure out who touched it last.
+			log_combat(transferred_by, target_organ.owner ? target_organ.owner : target_organ, "transferred reagents to", my_atom, "which had [get_external_reagent_log_string(transfer_log)]")
+		else
+			target_atom?.add_hiddenprint(transferred_by) //log prints so admins can figure out who touched it last.
+			log_combat(transferred_by, target_atom, "transferred reagents to", my_atom, "which had [get_external_reagent_log_string(transfer_log)]")
```
Comment (LemonInTheDark) [inline code/modules/reagents/chemistry/holder/holder.dm:542]: rather then duplicating the code like this, store as a var the thing we're transferring into, and set it in the isorgan case. much cleaner, easier to expand on in future.

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

File: code/modules/reagents/chemistry/holder/holder.dm (hunk 2 above)
Comment (lessthnthree) [reply code/modules/reagents/chemistry/holder/holder.dm:542]: yeah, that's nicer

File: code/modules/reagents/chemistry/holder/holder.dm [hunk 3]
```diff
@@ -532,9 +532,13 @@
...
 		transfer_log[reagent.type] = list(REAGENT_TRANSFER_AMOUNT = transfer_amount, REAGENT_PURITY = reagent.purity)
 
+	//combat log
 	if(transferred_by && target_atom)
+		if(isorgan(target_atom))
+			var/obj/item/organ/organ_item = target_atom
+			target_atom = organ_item.owner
 		target_atom.add_hiddenprint(transferred_by) //log prints so admins can figure out who touched it last.
-		log_combat(transferred_by, target_atom, "transferred reagents ([get_external_reagent_log_string(transfer_log)]) from [my_atom] to")
+		log_combat(transferred_by, target_atom, "transferred reagents to", my_atom, "which had [get_external_reagent_log_string(transfer_log)]")
```
Comment (LemonInTheDark) [inline code/modules/reagents/chemistry/holder/holder.dm:541]: Make a var called like "log_target" or something here, we shouldn't modify the value of target_atom outside this if check

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

File: code/modules/reagents/chemistry/holder/holder.dm [hunk 4]
```diff
@@ -532,9 +532,16 @@
...
+	//combat log
 	if(transferred_by && target_atom)
-		target_atom.add_hiddenprint(transferred_by) //log prints so admins can figure out who touched it last.
-		log_combat(transferred_by, target_atom, "transferred reagents ([get_external_reagent_log_string(transfer_log)]) from [my_atom] to")
+		var/atom/log_target
+		if(isorgan(target_atom))
+			var/obj/item/organ/organ_item = target_atom
+			log_target = organ_item.owner
+		else
+			log_target = target_atom
```
Comment (LemonInTheDark) [inline code/modules/reagents/chemistry/holder/holder.dm:542]:
```suggestion
		var/atom/log_target = target_atom
		if(isorgan(target_atom))
			var/obj/item/organ/organ_item = target_atom
			log_target = organ_item.owner
```

I prefer this pattern for this sort of filtering. Cleaner, more clear about the default IMO

Comment (LemonInTheDark) [reply code/modules/reagents/chemistry/holder/holder.dm:542]: Last review I promise

Comment (lessthnthree) [reply code/modules/reagents/chemistry/holder/holder.dm:542]: Yeah I agree that looks nicer. That was my original way doing it, thing is coming back full circle.

The null question is one I'm not entirely sure. Can you transfer a reagent to an organ that's outside of a body? Probably. Does it use this specific proc? (The only way I thought would be a syringe, and that doesn't use this one.)

Think it needs a null check?

idk I'll test it a bit more tomorrow

**Pull request merged**